csrc/executor_kernel_arg.cpp

```diff
@@ -88,36 +80,20 @@ std::unique_ptr<TensorArgAbstract> getTensorArg(int nDims) {
   return nullptr;
 }
 
+template <typename nvfuser_index_t>
+struct GetTensorArgWithNativeType {
+  template <typename T>
+  std::unique_ptr<TensorArgAbstract> operator()(int ndims) {
+    return getTensorArg<T, nvfuser_index_t>(ndims);
+  };
+};
+
 template <typename INDEX_MODE>
 std::unique_ptr<TensorArgAbstract> getTensorArg(
     c10::ScalarType dtype,
     int nDims) {
-  switch (dtype) {
-    case c10::ScalarType::Double:
-      return getTensorArg<double, INDEX_MODE>(nDims);
-    case c10::ScalarType::Float:
-      return getTensorArg<float, INDEX_MODE>(nDims);
-    case c10::ScalarType::Half:
-      return getTensorArg<at::Half, INDEX_MODE>(nDims);
-    case c10::ScalarType::BFloat16:
-      return getTensorArg<at::BFloat16, INDEX_MODE>(nDims);
-    case c10::ScalarType::Bool:
-      return getTensorArg<bool, INDEX_MODE>(nDims);
-    case c10::ScalarType::Long:
-      return getTensorArg<int64_t, INDEX_MODE>(nDims);
-    case c10::ScalarType::Int:
-      return getTensorArg<int32_t, INDEX_MODE>(nDims);
-    case c10::ScalarType::ComplexFloat:
-      return getTensorArg<c10::complex<float>, INDEX_MODE>(nDims);
-    case c10::ScalarType::ComplexDouble:
-      return getTensorArg<c10::complex<double>, INDEX_MODE>(nDims);
-    default:
-      TORCH_CHECK(
-          false,
-          "Dtype: ",
-          dtype,
-          " not currently supported in code generated kernels.");
-  }
+  return atenTypeDispatchWithC10Complex(
```

Comment:
I believe here we can just use `std::complex<float>` as we just pass pointers from host to device.